lib/puppet/type/archive.rb

```diff
@@ -103,7 +103,7 @@ def should_to_s(value)
   newparam(:source) do
     desc 'archive file source, supports http|https|ftp|file uri.'
     validate do |value|
-      unless value =~ URI.regexp(%w(http https file ftp))
+      unless value =~ URI.regexp(%w(http https file ftp)) || value =~ %r{^([A-Z]:|\/)}
```

Comment:
yes, do you know how stub os correctly to run rspec for Puppet::Util.absolute_path? otherwise spec tests will fail for Windows test cases.